pootle/apps/pootle_store/unit/filters.py

```diff
@@ -6,6 +6,139 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+from django.db.models import Q
+
+from pootle_statistics.models import SubmissionTypes
+from pootle_store.models import SuggestionStates
+from pootle_store.util import FUZZY, TRANSLATED, UNTRANSLATED
+
+
+class FilterNotFound(Exception):
+    pass
+
+
+class UnitFilter(object):
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+
+    def filter(self, unit_filter):
+        try:
+            return getattr(
+                self, "filter_%s" % unit_filter.replace("-", "_"))()
+        except AttributeError:
+            raise FilterNotFound()
+
+
+class UnitChecksFilter(UnitFilter):
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+        self.checks = kwargs.get("checks")
+        self.category = kwargs.get("category")
+
+    def filter_checks(self):
+        if self.checks:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__name__in=self.checks).distinct()
+        elif self.category:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__category=self.category).distinct()
+        return self.qs.none()
+
+
+class UnitStateFilter(UnitFilter):
+    """Filter a Unit qs based on unit state"""
```

Comment:
There is no `filter_all` method so when providing a `'all'` unit_filter an empty queryset will be returned, am I wrong?